File: frontend/angular/src/app/core/dtos/stock-lots/stock-lot-search-request.dto.ts
import { StockLotCondition } from '../shipping-receiving';

export interface StockLotSearchRequestDto {
  lotNumber?: string;
  checkedInFrom?: string;
  checkedInTo?: string;
  diameterExact?: number;
  diameterMin?: number;
  diameterMax?: number;
  condition?: StockLotCondition;
  sortColumn?: string;
  sortDirection?: string;
  page?: number;
  pageSize?: number;
}

File: backend/CncApp/CncApp.Application/Dtos/StockLots/StockLotSearchRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace CncApp.Application.Dtos.StockLots;

public class StockLotSearchRequestDto
{
    public string? LotNumber { get; set; }
    public DateTimeOffset? CheckedInFrom { get; set; }
    public DateTimeOffset? CheckedInTo { get; set; }
    public decimal? DiameterExact { get; set; }
    public decimal? DiameterMin { get; set; }
    public decimal? DiameterMax { get; set; }
    public string SortColumn { get; set; } = "CheckedInDateTime";
    public string SortDirection { get; set; } = "desc";

    [Range(1, int.MaxValue)]
    public int Page { get; set; } = 1;

    [Range(1, 100)]
    public int PageSize { get; set; } = 25;
}
